static-assets/components/cstudio-forms/controls/node-selector.js

```diff
@@ -534,12 +534,12 @@ YAHOO.extend(CStudioForms.Controls.NodeSelector, CStudioForms.CStudioFormField,
 		return validation;
 	},
 
-	newInsertItem: function (key, value, type) {
+	newInsertItem: function (key, value, type, datasource) {
```

Comment:
When inserting a new item to the node-selector, there's this logic where we set the datasourceId to the item, so when editing an item of the datasource we read the datasourceId to know what data source will be used for edition.
The data source prop got lost in `newInsertItem`, causing the default datasource (first in list of datasources) to be used for edition, ending up in errors in some scenarios